search-meter.php

```diff
@@ -340,14 +353,13 @@ function tguy_sm_save_search($posts) {
 		// Save search summary into the DB. Usually this will be a new row, so try to insert first
 		// Temporarily suppress errors, as this query is expected to fail on duplicate searches in a single day. Thanks to James Collins.
 		$suppress = $wpdb->suppress_errors();
-		$success = $wpdb->query($wpdb->prepare("
-			INSERT INTO `{$wpdb->prefix}searchmeter` (`terms`,`date`,`count`,`last_hits`)
-			VALUES (%s, UTC_DATE(), 1, %d)",
-			$search_terms,
-			$hit_count
-		));
-		$wpdb->suppress_errors($suppress);
-		if (!$success) {
+
+		$rowcount = $wpdb->get_var($wpdb->prepare(
```

Comment:
I would prefer to use INSERT... ON DUPLICATE KEY UPDATE, which will be much simpler. Should have done it years ago. Just have to append `ON DUPLICATE KEY UPDATE count = VALUES(count), last_hits = VALUES(last_hits)` to the INSERT query, and remove the error check and UPDATE. I will do that when I have a few minutes for testing, or you can do it if you're keen.